### DISCUSSION TITLE
Lock down our repo from bots

### DISCUSSION THREAD

**Discussion by tcharding** [rust-bitcoin/rust-bitcoin#3124] [General]
We are getting a few bots lately and they trigger notifications which is annoying. Perhaps we should lock down the repo a bit. Seems we can do so by following this doc: https://docs.github.com/en/communities/moderating-comments-and-conversations/limiting-interactions-in-your-repository

Posting as a discussion in case there are better strategies.

For an example see: https://github.com/rust-bitcoin/rust-bitcoin/pull/2736 (bot account ErnestoLara approves merged branch)

**Comment by tcharding** [comment]
I wonder if "limit to prior contributors" stops bots from approving things?

**Comment by Kixunil** [comment]
I don't think there's any solution. Blacklists don't work an whitelists are hostile towards external contributors. Also I don't perceive the problem to be significant. If we get ton of spam we could consider implementing an LN-based spam filter - a new contributor pays a deposit over LN and if his contribution is found to not be a spam it's returned. Or alternatively make the payment very small to just undermine the non-targeted spam bots.

**Reply by storopoli** [reply]
I also think that any solution would be non-trivial to implement and would create a big barrier to external/new contributors.
We just got spammed now (I think 3 or 4 approvals right?), when was the last time this happened? I just want to measure how of a menace the bots are.

**Comment by apoelstra** [comment]
As others have said, I think barriers will be too much of an imposition to new contributors (and likely some existing ones, if we don't make our whitelist big enough). I think the bots are something we'll have to live with.

**Comment by tcharding** [comment]
No worries, cheers. (Keep calm and ignore bots).
